hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -41,6 +42,8 @@
 
 	private Statement lastQuery;
 
```

Comment:
The `currentBatch` depends on a particular `Statement`. When that `Statement` is released (i.e, by `ResourceRegistryStandardImpl#release(Statement)`, then we will still have `currentBatch` in `ResourceRegistryStandardImpl`.

It seems inconsistent, but I'm not sure if it will cause any problems.